### PR TITLE
Problem: not able to use wepoll in configure script

### PR DESCRIPTION
**Pull request by pijyoi**
Solution: add wepoll as a polling system

### CODE REVIEW

**Comment by sigiesec** [comment]
I wasn't aware that the autotoola build can be used on Windows. With what toolset are you using it? Should we have a CI job for that maybe?

**Comment by pijyoi** [comment]
I was cross compiling from Fedora using distro provided mingw64 tool chain.

It's also possible to use the configure script in Windows under msys
environment. Although my preference then would be to use cmake.

**Comment by sigiesec** [comment]
Ah ok... That makes sense. I was not thinking about MinGW when adding wepoll. 

We should probably have a CI job for MinGW/MSys, at least using either AutoTools or CMake, maybe both, or declare the other unsupported. I made up a ticket some time ago: https://github.com/zeromq/libzmq/issues/3031
If you wanted to do that, to ensure it doesn't break, that would be great :)

I started work on doing the same for Cygwin64: https://github.com/sigiesec/libzmq/tree/add-cygwin64-ci, but did not finish that yet. Some tests are failing, and I need to investigate why.

File: Makefile.am [hunk 1]
```diff
@@ -257,6 +257,11 @@ src_libzmq_la_SOURCES = \
 	src/zap_client.hpp \
 	src/zmq_draft.h
 
+if USE_WEPOLL
+src_libzmq_la_SOURCES += \
+	external/wepoll/wepoll.c
```
Comment (bluca) [inline Makefile.am]: the header needs to be included too, or it won't be part of the dist tarball

**Comment by bluca** [comment]
If it needs to be used with autotools, the other files in the wepoll directory (especially the license) need to be distributed too (EXTRA_DIST in Makefile.am)

**Pull request merged**